src/libraries/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.Shared.xml

```diff
@@ -29,7 +29,11 @@
     <type fullname="System.Runtime.CompilerServices.AsyncVoidMethodBuilder">
       <property name="ObjectIdForDebugger" />
     </type>
-    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder*">
+    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder">
+      <property name="ObjectIdForDebugger" />
+      <method name="SetNotificationForWaitCompletion" />
+    </type>
+    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder&lt;TResult&gt;">
```

Comment:
```suggestion
    <type fullname="System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1">
```

This change actually fixes this warning when linking coreclr's System.Private.CoreLib:

```
resource System.Private.CoreLib.xml in System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e : error IL2008: Could not resolve type 'System.Runtime.CompilerServices.AsyncTaskMethodBuilder<TResult>' specified in resource System.Private.CoreLib.xml in System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e [D:\repos\dotnet_runtime_2\src\coreclr\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```